Getting started: registering (#57)
codewars/docs: content/getting-started/registering.md

next: "/getting-started/setting-up/"
---

<!--
Steps
1. Sign up
2. Choose language
3. Multiply CAPTCHA
4. Enlist
 - confirmation email
5. Initial training setup
6. Initial dashboard, popup "Let's get started!", read more
-->

# Registering

So you decided to join? That's great! There's only a couple of steps to create new account and start solving tasks, learn, and improve your skills!

## First step

First, you need to go to [Sign Up](https://www.codewars.com/join) page, if you haven't yet. You will be presented with a board of many languages to choose from:

![languages board](./img/registering_01_choose_language.png)

Choose whatever language you like, you want to learn, or are familiar with. I want to improve my skills with functional programming, and I already know a bit if Java, so I choose Scala as my initial language. Here you can pick any language available, your selection does not determine anything yet, and you will be always able to choose another language later, if you wish.

## Your first challenge

> After selecting the language, you are presented with your very first challenge. It's an entry level task which you need to solve to prove your dedication and commitment to learning, and prove yourself worthy of training among CodeWarriors. It should not be difficult, let's see...

Reviewer: ```suggestion
After selecting the language, you are presented with your very first challenge. It's an entry level task which you need to solve to prove your dedication and commitment to learning.
```

"It should not be difficult" was already said with "an entry level task".

We can omit "prove yourself worthy of training among CodeWarriors" since it can be misinterpreted as unwelcoming and `codewarrior` is a new term.

By the way, we actually don't really like the term `Codewarrior`. Jake told me one of the early contributors named it and he doesn't like it. We should probably stop using it or reduce its usage.

Author: Apparently you haven;t tried to solve that 8 kyu AGDA kata :)

I thought that "entry level" != "easy", especially for newbies, and I wanted to sound encouraging. But if you say so, than ok 👍

![first task](./img/registering_02_multiply_initial.png)

Task says: "The code does not execute properly. Try to figure out why." Let's take a closer look at it... Of course! Some tiny detail is missing, and it causes compilation error! Let's put it in place, fix the code, and click `SUBMIT`. Server will process our solution and... after a short while, tests report is presented to you. If you made a mistake and your answer is not correct, don't worry! Just click `TRY AGAIN` and fix your solution.

![retry](./img/registering_03_multiply_wrong_answer.png)

If your answer is correct, you will experience the best little thing in a life of a coder _[citation needed]_: **green tests**! Congratulations!

![success](./img/registering_04_multiply_success.png)

## Your new account

> You have proved yourself worthy. You have proved yourself committed and willing to learn and improve your skills. You are allowed to enter Codewars dojo. To get your account created, fill in a couple of details, or use your GitHub account to authenticate on Codewars:

Reviewer: > You have proved yourself worthy. You have proved yourself committed and willing to learn and improve your skills. You are allowed to enter Codewars dojo.

Is too repetitive.

![enlist](./img/registering_05_enlist.png)

Great, your account is created!

**Important note:** Codewars will send you an email to confirm your email account. Go to your inbox and click on confirmation link in the message titled `"Confirm your account... Dojo Access Link"`, received from Codewars. Until you do this, you will be allowed to train and solve tasks, but some actions will be restricted or unavaliable for you. For example, you won't be able to create discourse posts. If you cannot find the email, check `Spam` folder of your inbox! If you do not confirm your account right now, you can always ask Codewars to resend you the confirmation email by clicking dedicated button in your account panel (we will get to your account panel soon).

> ## Your training routine

Reviewer: This section should go to "Setting Up".

Author: I was thinking the same while writing,gave it a bit of thought, and decided to keep it here deliberately. Thing is that initial training setup is a part of sign-up flow and a user cannot skip it or navigate out of it. It feels as integral part of sign up, as opposed to setting up your account after it's created. Having it here lets a new user to go through all the sign up process while reading a single page. Moving it to some other page will make user navigate to other place even before process feels finished and they are able to use their account.
On the other hand, training setup would have to be somehow covered, or referenced, in Setting Up part, and parts of the description can turn out to be redundant. But until I have that page ready and will be able how/if it can be referenced or mentioned, I would like to keep it here. Is this OK?

Reviewer: Hmm, I think mentioning that the initial training setup exists and having a link to more details (next page) is natural, but we can improve it later.

> But until I have that page ready and will be able to see how/if it can be referenced or mentioned, I would like to keep it here. Is this OK?

Sure

next: "/getting-started/setting-up/"
---

<!--
Steps
1. Sign up
2. Choose language
3. Multiply CAPTCHA
4. Enlist
 - confirmation email
5. Initial training setup
6. Initial dashboard, popup "Let's get started!", read more
-->

# Registering

So you decided to join? That's great! There's only a couple of steps to create new account and start solving tasks, learn, and improve your skills!

## First step

First, you need to go to [Sign Up](https://www.codewars.com/join) page, if you haven't yet. You will be presented with a board of many languages to choose from:

![languages board](./img/registering_01_choose_language.png)

> Choose whatever language you like, you want to learn, or are familiar with. I want to improve my skills with functional programming, and I already know a bit if Java, so I choose Scala as my initial language. Here you can pick any language available, your selection does not determine anything yet, and you will be always able to choose another language later, if you wish.

Reviewer: > Your selection does not determine anything yet

I think we need to explain that the user will need to solve a simple challenge with the chosen language.

---

I understand the intention of including "I want to improve my skills ...", but it seems out of place and can be confusing.

![enlist](./img/registering_05_enlist.png)

Great, your account is created!

**Important note:** CodeWars will send you an email to confirm your email account. Go to your inbox and click on confirmation link in the message titled `"Confirm your account... Dojo Access Link"`, received from CodeWars. Until you do this, you will be allowed to train and solve tasks, but some actions will be restricted or unavaliable for you. For example, you won't be able to create discourse posts. If you cannot find the email, check `Spam` folder of your inbox! If you do not confirm your account right now, you can always ask CodeWars to resend you the confirmation email by clicking dedicated button in your account panel (we will get to your account panel soon).

## Your training routine

There's only one last step you need to do before you start solving tasks: you need to set up your training routine, so CodeWars would know what to suggest you and how to drive you through your learning process. Don't worry, you just need to answer a few simple questions:

![initial training setup](./img/registering_06_initial_training_setup.png)

>  - **What is your programming experience?** - select any option which you feel describes you the best. If you are making your first steps in the world of programming, just pick "Learning to Program". If you are a seasoned programmer who wants to train their coding-fu, improve their skills, and probably learn somethng new, you can choose "Senior Developer".

Reviewer 2: - **What is your programming experience?** - select any option which you feel describes you the best. If you are making your first steps in the world of programming, just pick "Learning to Program". If you are a seasoned programmer who wants to train their coding-fu, improve their skills, and probably learn something new, you can choose "Senior Developer".

>  - **Would you like to receive our newsletter with weekly challenges?** - choose "Yes" to receive a weekly email with some suggested tasks to solve. Email contains a choice of CodeWars problems of every difficulty, so you will be always able to find something what would suit you. Participating in Weekly Coding Challenge is a great way to keep yourself motivated! If you do not want to receive the email, just pick "No".

Reviewer 2: - **Would you like to receive our newsletter with weekly challenges?** - choose "Yes" to receive a weekly email with some suggested tasks to solve. Email contains a choice of Codewars problems of any difficulty, so you will always be able to find something that should suit you. Participating in Weekly Coding Challenge is a great way to keep yourself motivated! If you do not want to receive the email, just pick "No".

>  - **Choose the language you wish to train on** - you can select from a large list of languages available on CodeWars. You can select one, but if you want, you can select more. During your training, CodeWars suggests you tasks which are available in languages you choose here. If you are not sure if you want to select some language or not, don't worry! You can always switch to any language available for trained task. As I want to improve my functional programming skills, and as I heard, Haskell is a very nice FP language, I am going to select this one too.

Reviewer 2: - **Choose the language you wish to train on** - you can select from a large list of languages available on Codewars. You can select one, but if you want, you can select more. During your training, Codewars suggests you (through the [dashboard](https://www.codewars.com/dashboard)) tasks which are available in languages you choose here. If you are not sure if you want to select some language or not, don't worry! You can always switch to any language available for trained task. As I want to improve my functional programming skills, and as I heard, Haskell is a very nice FP language, I am going to select this one too.

> Remember that if you would like to change your answers later, you will be always able to do so in your [Training Setup panel](https://www.codewars.com/trainer/setup). If you answered all above questions, just click `SAVE` and you are done!

Reviewer 2: Remember that if you would like to change your answers later, you will always be able to do so in your [Training Setup panel](https://www.codewars.com/trainer/setup). If you answered all above questions, just click `SAVE` and you are done!

## Your first challenge

It's an entry level task which you need to solve to prove your dedication and commitment to learning. Let's see...

![first task](./img/registering_02_multiply_initial.png)

Task says: "The code does not execute properly. Try to figure out why." Let's take a closer look at it... Of course! Some tiny detail is missing, and it causes compilation error! Let's put it in place, fix the code, and click `SUBMIT`. Server will process our solution and... after a short while, tests report is presented to you. If you made a mistake and your answer is not correct, don't worry! Just click `TRY AGAIN` and fix your solution.

![retry](./img/registering_03_multiply_wrong_answer.png)

If your answer is correct, you will experience the best little thing in a life of a coder _[citation needed]_: **green tests**! Congratulations!

![success](./img/registering_04_multiply_success.png)

## Your new account

You have solved your first Codewars challenge. Great! Now you are ready to have your account created. Just fill in a couple of details, or use your GitHub account to authenticate on Codewars:

![enlist](./img/registering_05_enlist.png)

Great, your account is registered!

> **Important note:** Codewars will send you an email to confirm your email account. Go to your inbox and click on confirmation link in the message titled `"Confirm your account... Dojo Access Link"`, received from Codewars. Until you do this, you will be allowed to train and solve tasks, but some actions will be restricted or unavaliable for you. For example, you won't be able to create discourse posts. If you cannot find the email, check `Spam` folder of your inbox! If you do not confirm your account right now, you can always ask Codewars to resend you the confirmation email by clicking dedicated button in your account panel (we will get to your account panel soon).

Reviewer: ```suggestion
**Important note:** Codewars will send you an email asking to confirm the account. Some actions like posting comments will be restricted until you do. If you cannot find the email in your inbox, make sure to check the spam folder as well. The button to resend the confirmation email is shown in the Account Settings if your account is not confirmed. We'll show you where to find the Account Settings soon.
```

I don't think the email title should be in the documentation and how to confirm is in the email.

I also cleaned up some wording. "create discourse posts" is unnecessarily indirect. We'll probably stop using "Discourse" for internal comment system because we'll be adding the Discourse forum to avoid confusion.

---

I might add some Markdown extension for these notes/warnings to display in some consistent way.

## Next steps

> You have successfully created your account and registered it with Codewars. You are strongly recommended to [set up](./setting-up/) your account and training routine, but if you wish, you can go straight to your [dashboard](./my-profile/) or even find your [next challenges to solve](./finding-kata/).

Reviewer: ```suggestion
You have successfully created your account and registered it with Codewars. You are strongly recommended to [set up](/getting-started/setting-up/) your account and training routine, but if you wish, you can go straight to your [dashboard](./my-profile/) or even find your [next challenges to solve](/getting-started/finding-kata/).
```

Where should `./my-profile` go?

Author: I restored link to `dashboard` in this place, idea of `my-profile` needs more thought.
